src/UserRepository.js

```diff
@@ -0,0 +1,21 @@
+class UserRepository {
+  constructor(users) {
+    this.allUsers = users;
```

Comment:
**suggestion (non-blocking)**: I would think that a user repository holds all the users, so perhaps `users` is a better property name than `allUsers`.